src/api/endpoints/auth/TokenManager.ts

```diff
@@ -37,17 +38,22 @@ export default class TokenManager extends ApiEndpoint {
     // if the refresh token isn't present, nothing we can do
     if (this.refreshToken === undefined) return this.accessToken;
 
-    // token is expired (or will expire soon), so fetch a new one
-    const tokens = await this.fetchToken(this.refreshToken);
+    // token is expired (or will expire soon) and there is not already one fetching, so fetch a new one
+    if (!this.tokenFetchPromise) this.tokenFetchPromise = this.fetchToken(this.refreshToken);
```

Comment:
this isn't atomic so I guess there's still technically a small chance this is entered multiple times, but I'm also not aware of any way to lock atomically in javascript :/

Reply:
Yeah the native one with js only works with buffers to my knowledge